consensus/istanbul/backend/backend.go

```diff
@@ -195,6 +193,20 @@ func New(config *istanbul.Config, db ethdb.Database) consensus.Istanbul {
 		}
 	}
 
+	announceConfig := AnnounceManagerConfig{
+		IsProxiedValidator: backend.IsProxiedValidator(),
+		AWallets:           &backend.aWallets,
+		VcDbPath:           config.VersionCertificateDBPath,
+	}
+
+	backend.announceManager = NewAnnounceManager(
+		announceConfig,
+		backend,
+		backend,
+		backend,
+		backend.valEnodeTable,
```

Comment:
thinking that maybe makes sense for the announce manager to OWN the valEnodeTable. Basically, the purpose of the announce protocol, is to:
- disclose our own enode url to other address
- maintain a table of others validators enodes (valEnodeTable)

so i can be represented as this "interface" where i go to fectch validator enodes, and that hooks into part of the system to do the announcing

Reply:
Yes, in fact that's what I wanted at first. However it's being used in a lot of places, so I want to be more careful with it. I'd like to first have a better implementation of the AnnounceManager before venturing into fully moving this one